homeassistant/components/minecraft_server/__init__.py

```diff
@@ -58,11 +66,110 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
     # Clean up.
     server.stop_periodic_update()
-    hass.data[DOMAIN].pop(unique_id)
+    hass.data[DOMAIN].pop(config_entry_id)
 
     return unload_ok
 
 
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old config entry to a new format."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    # 1 --> 2: Use config entry ID as base for unique IDs.
+    if config_entry.version == 1:
+        assert config_entry.unique_id
+        assert config_entry.entry_id
+        old_unique_id = config_entry.unique_id
+        config_entry_id = config_entry.entry_id
+
+        # Migrate config entry.
+        _LOGGER.debug("Migrating config entry. Resetting unique ID: %s", old_unique_id)
+        config_entry.unique_id = None
+        config_entry.version = 2
+        hass.config_entries.async_update_entry(config_entry)
+
+        # Migrate device.
+        await _async_migrate_device_identifiers(hass, config_entry, old_unique_id)
+
+        # Migrate entities.
+        await er.async_migrate_entries(hass, config_entry_id, _migrate_entity_unique_id)
+
+    _LOGGER.info("Migration to version %s successful", config_entry.version)
```

Comment:
Debug level please. The user doesn't need to know about config entry versions.

Reply:
I'll change it to debug level, no problem.
FYI: I took the info level from the example in the dev docs, maybe it should be updated to debug level as well: https://developers.home-assistant.io/docs/config_entries_config_flow_handler#config-entry-migration

Reply:
https://github.com/home-assistant/developers.home-assistant/pull/1901